gen.go

```diff
@@ -1853,21 +1863,28 @@ func (t *{{ .Name}}) UnmarshalCBOR(r io.Reader) (err error) {
 		return fmt.Errorf("{{ .Name }}: map struct too large (%d)", extra)
 	}
 
-	var name string
 	n := extra
 
+	nameBuf := make([]byte, {{ .MaxMapKeyLength }})
 	for i := uint64(0); i < n; i++ {
+		nameLen, ok, err := cbg.ReadFullStringIntoBuf(cr, nameBuf, {{ MaxLen 0 "String" }})
+		if err != nil {
+			return err
+		}
+
+		if !ok {
+			// Field doesn't exist on this type, so ignore it
+			cbg.ScanForLinks(cr, func(cid.Cid){})
```

Comment:
Do we not have a better way to ignore fields? Honestly, I'd special-case `Deferred` into an `Ignored` type that just drops everything.

Reply:
Also, we need to handle errors.

Reply:
yeah thats the way we've been doing it so far, probably makes sense to have a more dedicated thing for it

Reply:
Eh, this is probably fine (assuming you check for errors). It ensures that any CIDs contained in such fields are valid.